src/components/battle/MapListModal.vue

```diff
@@ -0,0 +1,33 @@
+<template>
+    <Modal ref="modal" :title="title" width="600px" height="75vh" padding="0">
+        <div class="container">
+            <MapListComponent @map-selected="mapSelected" />
+        </div>
+    </Modal>
+</template>
+
+<script lang="ts" setup>
+import { Ref, ref } from "vue";
+
+import Modal from "@/components/common/Modal.vue";
+import MapListComponent from "@/components/maps/MapListComponent.vue";
+import { MapData } from "@/model/map-data";
+
+const modal: Ref<null | InstanceType<typeof Modal>> = ref(null);
```

Comment:
This is only if you need to access the modal element in code, which isn't happening here, so can be safely removed